src/mapreduce.jl

```diff
@@ -159,27 +159,46 @@ end
 (-)(A::DArray, x::Number) = A .- x
 (-)(x::Number, A::DArray) = x .- A
 
-map_localparts(f::Callable, d::DArray) = DArray(i->f(localpart(d)), d)
-map_localparts(f::Callable, d1::DArray, d2::DArray) = DArray(d1) do I
-    f(localpart(d1), localpart(d2))
+function map_localparts(f::Callable, d::DArray)
+    r = Base.asyncmap!(similar(procs(d), Future), procs(d)) do p
+        remotecall_wait(p) do
+            f(localpart(d))
+        end
+    end
+    return DArray(r)
```

Comment:
`reduce_localparts` and `mapreduce_localparts` are fine. 

`localpart(mapreduce, abs2, +, A)` does not appear to read as "call `mapreduce` with `abs2` and `+` on the localpart of `A`" to me.